### PR TITLE
Feature/herolab patch note parser

### PR DESCRIPTION
**Pull request by Surxe**
Hero Lab patch notes are now parsed through their location in the gamefiles, rather than added to input-data manually.

* Removed the comments and readme for adding to input-data.
* Reformatted dates from forum changelogs from mm-dd-yyyy to yyyy_mm_dd, which are now consistent with gamefile date formats. This allows them to be sorted by date accurately in `changelog_configs.json` (formerly named `changelogs.json`).
* Fixed the TypedDicts accidentally describing the final format that `parse_changelogs` uses with Tagging, as opposed to it's actual format.

Still feel's a little scrappy, so lmk if you see any immediate areas of improvement. I'll need this merged in before I work on integrating the manifest-id versioning branch with the recent changelog updates.

_Parsed data in [deadlock-data PR](https://github.com/deadlock-wiki/deadlock-data/pull/25)_

### CODE REVIEW

File: src/changelogs/fetch_changelogs.py [hunk 1]
```diff
@@ -101,8 +93,96 @@ def _fetch_update_html(self, link):
...
+                # Extract the description
+                description = remaining_str[li_start + len('<li>') : li_end]
+
+                # Remove the description from the remaining value
+                remaining_str = remaining_str[li_end + len('<li>') :]
+
+                # Remove any remaining tags
+                tags_to_remove = ['<ul>', '</ul>', '<b>', '</b>', '<i>', '</i>']
+                for tag in tags_to_remove:
+                    description = description.replace(tag, '')
```
Comment (hassayag) [inline src/changelogs/fetch_changelogs.py:168]: this is a little hard to follow, so worth shoving in its own function

File: src/changelogs/fetch_changelogs.py [hunk 2]
```diff
@@ -152,6 +233,21 @@ def _process_local_changelogs(self, changelog_path):
...
+    """
+    if '/' in date:
+        date = date.split('/')
+    elif '-' in date:
+        date = date.split('-')
+    else:
+        raise ValueError(f'Invalid date format {date}')
+
+    date = f'{date[2]}_{date[0]}_{date[1]}'
+    return date
```
Comment (hassayag) [inline src/changelogs/fetch_changelogs.py:253]: One of the dates is `2024_11_1`. Is that a fault of the raw file?

File: src/changelogs/fetch_changelogs.py [hunk 3]
```diff
@@ -19,78 +19,70 @@ class Changelog(TypedDict):
...
     """
 
-    # Hero lab changelogs need to be added manually to
-    # ./input-data/raw-changelogs following the
-    # naming convention 'herolab_2024_10_29.txt'
-    # see the referenced file for example formatting
+    def __init__(self, client_version, update_existing):
+        self.changelogs: dict[str, ChangelogString] = {}
+        self.changelog_configs: dict[str, ChangelogConfig] = {}
+        self.client_version = client_version
```
Comment (hassayag) [inline src/changelogs/fetch_changelogs.py:36]: this is unused

**Review by hassayag — CHANGES_REQUESTED** [review]

File: src/changelogs/fetch_changelogs.py (hunk 2 above)
Comment (Surxe) [reply src/changelogs/fetch_changelogs.py:253]: This is the decompiled line: ```
"Citadel_PatchNotes_HeroLabs_hero_slork_3": "<b>11/1/2024</b>\t\t\t<li>Health per boon reduced from 41 to 36\t\t\t<li>Shotgun damage per pellet reduced from 4.5 to 4\t\t\t<li>Shotgun falloff start reduced from 18m to 16m\t\t\t<li>Scalding Spray weapon damage gain per sec reduced from 8% to 7%\t\t\t<li>Fathom's Bloodletting updated to have the same values as the latest version of Shiv's Bloodletting</li>",
    ```
    So I think it's correct. Recall its yyyy_mm_dd

File: src/changelogs/fetch_changelogs.py [hunk 4]
```diff
@@ -101,8 +93,96 @@ def _fetch_update_html(self, link):
...
+            header = f'[ HeroLab {hero_name_en} ]'
+
+            # Initialize the changelog entry
+            first_line_of_entry = False
+            if raw_changelog_id not in gamefile_changelogs:
+                gamefile_changelogs[raw_changelog_id] = ''
+                first_line_of_entry = True
+
+            # Add the header to the changelog entry
+            gamefile_changelogs[raw_changelog_id] += (not first_line_of_entry) * '\n' + header + '\n'
```
[the remark this answers is not in the document]
Comment (Surxe) [reply src/changelogs/fetch_changelogs.py]: I don't see a reasonable way to do this; best I can think of is `raw_changelog_id, date = self._initialize_log_line(key, value, gamefile_changelogs)` but feels a bit bloated. Instead I extracted some smaller bits to other functions (and below), and elaborated more in the comments. Lmk if you got a particular idea in mind.

File: src/changelogs/fetch_changelogs.py (hunk 3 above)
Comment (Surxe) [reply src/changelogs/fetch_changelogs.py:36]: Ignore `client_version` for now, it's in preparation of manifest-id versioning. Consider it a reminder for our future selves :wink:

Comment (hassayag) [reply src/changelogs/fetch_changelogs.py:36]: I know you're half joking, but if its a necessary part of the manifest feature, create a git issue with that information instead of unused code

File: src/changelogs/fetch_changelogs.py (hunk 2 above)
Comment (hassayag) [reply src/changelogs/fetch_changelogs.py:286]: Sure its valve's fault. Can we have it read `2024_11_01`.

Comment (Surxe) [reply src/changelogs/fetch_changelogs.py:294]: Ah I couldn't tell that's what you meant originally. Pushed an addition to `format_date`.

File: src/changelogs/fetch_changelogs.py (hunk 3 above)
Comment (Surxe) [reply src/changelogs/fetch_changelogs.py:36]: Aye sounds good. If I'm being serious, I had added it pre-emptively and forgotten that it would actually require the manifest-versioning branch until you mentioned it here lol.

**Comment by Surxe** [comment]
Reverted to an older version and ported to PR #130
